@here/harp-mapview/lib/DecodedTileHelpers.ts

```diff
@@ -422,65 +491,239 @@ export function getMaterialConstructor(technique: Technique): MaterialConstructo
+            material[propertyName as keyof THREE.Material],
+            techniqueAttrValue,
+            level
+        );
+    } else {
+        m[propertyName] = evaluateProperty(techniqueAttrValue, level);
+    }
+}
+
+/**
+ * Apply technique color to material taking special care with transparent (RGBA) colors.
+ *
+ * @note This function is intended to be used with secondary, triary etc. technique colors,
+ * not the base ones that may contain transparency information. Such colors should be processed
+ * with [[applyTechniqueBaseColorToMaterial]] function.
+ *
+ * @param technique an technique the applied color comes from
+ * @param material the material to which color is applied
+ * @param prop technique property (color) name
+ * @param value color value
+ * @param level optional, tile zoom level for zoom-level dependent properties are evaluated.
+ */
+export function applySecondaryColorToMaterial(
+    materialColor: THREE.Color,
+    techniqueColor: Value | Expr | InterpolatedProperty,
+    level?: number
+) {
+    let value = evaluateColorProperty(techniqueColor, level);
+
+    if (ColorUtils.hasAlphaInHex(value)) {
+        logger.warn("Used RGBA value for technique color without transparency support!");
+        // Just for clarity remove transparency component, even if that would be ignored
+        // by THREE.Color.setHex() function.
+        value = ColorUtils.removeAlphaFromHex(value);
+    }
+
+    materialColor.setHex(value);
+}
+
+/**
+ * Apply technique base color (transparency support) to material with modifying material opacity.
+ *
+ * This method applies main (or base) technique color with transparency support to the corresponding
+ * material color, with an effect on entire [[THREE.Material]] __opacity__ and __transparent__
+ * attributes.
+ *
+ * @note Transparent colors should be processed as the very last technique attributes,
+ * since their effect on material properties like [[THREE.Material.opacity]] and
+ * [[THREE.Material.transparent]] could be overridden by corresponding technique params.
+ *
+ * @param technique an technique the applied color comes from
+ * @param material the material to which color is applied
+ * @param prop technique property (color) name
+ * @param value color value in custom number format
+ * @param level optional, tile zoom level for zoom-level dependent properties are evaluated.
+ */
+export function applyBaseColorToMaterial(
+    material: THREE.Material,
+    materialColor: THREE.Color,
+    technique: Technique,
+    techniqueColor: Value,
+    level?: number
+) {
+    const colorValue = evaluateColorProperty(techniqueColor, level);
+
+    const { r, g, b, a } = ColorUtils.getRgbaFromHex(colorValue);
+    // Override material opacity and blending by mixing technique defined opacity
+    // with main color transparency
+    const tech = technique as any;
+    let opacity = a;
+    if (tech.opacity !== undefined) {
+        opacity *= evaluateProperty(tech.opacity, level);
```

Comment:
Is all this supposed to work if opacity or the base color are dynamic properties?

Reply:
As far as I can see it is (i.e. applyBaseColorToMAterial is called in the onBeforeRender callbacks).
Please rename level to zoomLevel to make it more clear.

Reply:
Fixed.